service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerTest.java

```diff
@@ -282,6 +303,76 @@ public void cloneWorkspace() throws Exception {
         sourceWorkspace.getDisplayName() + " (Copy)", destinationWorkspace.getDisplayName());
   }
 
+  @Test
+  public void cloneRawlsWorkspace() throws Exception {
+    UUID sourceWorkspaceId = UUID.randomUUID();
```

Comment:
I think it makes tests easier to read if you use helper methods. Also, future tests can reuse those helper methods.

```
  @Test
  public void cloneRawlsWorkspace() throws Exception {
    // Create source RAWLS workspace
        mockMvcUtils.createWorkspace(RAWLS);

    // Create some data repo references
    ApiDataRepoSnapshotResource snap1 =
        mockMvcUtils.createDataRepoSnapshotReference(USER_REQUEST, sourceWorkspaceId);
    ApiDataRepoSnapshotResource snap2 =
        mockMvcUtils.createDataRepoSnapshotReference(USER_REQUEST, sourceWorkspaceId);

    // Clone the rawls workspace into a destination rawls workspace
    // This relies on the mocked Sam check
    UUID destinationWorkspaceId = UUID.randomUUID();
    ApiCloneWorkspaceResult clonedWorkspace = mockMvcUtils.cloneWorkspace();

    List<ApiResourceCloneDetails> cloneDetails = cloneWorkspace.getWorkspace().getResources();
    assertEquals(2, cloneDetails.size());
    List<UUID> cloneIdList =
        cloneDetails.stream().map(ApiResourceCloneDetails::getSourceResourceId).toList();
    assertThat(cloneIdList, hasItem(snap1.getMetadata().getResourceId()));
    assertThat(cloneIdList, hasItem(snap2.getMetadata().getResourceId()));
  }
```

mockMvcUtils already has create createWorkspaceWithCloudContext.

I see you added mockMvcUtils.getCloneWorkspaceResult(). It would remove more boilerplate if the helper include the entire clone call -- everything from initial request to the stuff in getCloneWorkspaceResult().

Reply:
I'm going to send a PR to clean up all tests in this file. If you want, I can do this in my PR.

Reply:
I'm in the process of cleaning up the mock as you suggested. I'll send it past you for a check before merging.